src/TMS/CourseCreation/ChildAssistant.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace ILIAS\TMS\CourseCreation;
+
+trait ChildAssistant
+{
+	/**
+	 * Get all children by type recursive
+	 *
+	 * @param int 	$ref_id
+	 * @param string 	$search_type
+	 *
+	 * @return Object 	of search type
+	 */
+	protected function getAllChildrenOfByType($ref_id, $search_type)
+	{
+		$g_tree = $this->getDIC()->repositoryTree();
+		$g_objDefinition = $this->getDIC()["objDefinition"];
+		$childs = $g_tree->getChilds($ref_id);
+		$ret = array();
+
+		foreach ($childs as $child) {
+			$type = $child["type"];
+			if ($type == $search_type) {
+				$ret[] = \ilObjectFactory::getInstanceByRefId($child["child"]);
+			}
+
+			if ($g_objDefinition->isContainer($type)) {
+				$rec_ret = $this->getAllChildrenOfByType($child["child"], $search_type);
```

Comment:
Ich finde das hier ließe sich in Nu in eine Iteration umwandeln, geht aber auch ohne.
Der Vorteil: dann müsstest du auch $g_tree und g_objDefinition nur einmal holen...

Reply:
Ja stimmt. Ich kann im Moment aber aus Zeitgründen nicht Garantieren, dass es diese Globalen überall gibt. Deshalb müssen wir sie mehrmals holen.